src/arizona_template_compiler.erl

```diff
@@ -0,0 +1,653 @@
+    end.
+
+block_assigns(Attrs, Macros, State) ->
+    maps:from_list(lists:map(fun
+        ({K, {text, _, Txt}}) ->
+            {binary_to_existing_atom(K), {text, Txt}};
+        ({K, {expr, Loc, Expr}}) ->
+            Eval = should_eval_expr(Expr, Macros),
+            {binary_to_existing_atom(K), expand_expr(Expr, Loc, Eval, State)}
+    end, Attrs)).
+
+block_assigns_macros(Assigns) ->
+    #{K => Txt || K := {text, Txt} <- Assigns}.
+
+norm_block_assigns(Assigns) ->
+    #{K => maps:with([function, vars], Expr) || K := {expr, Expr} <- Assigns}.
+
+should_eval_expr(Expr, Macros) when map_size(Macros) > 0 ->
+    ExprTree = merl:quote(Expr),
+    ExprVars = merl:template_vars(merl:template(ExprTree)),
+    ExprVars =/= [] andalso
+        lists:all(fun(Var) -> is_map_key(Var, Macros) end, ExprVars);
+should_eval_expr(_, _) ->
+    false.
+
+block(Tree0, NormAssigns, State) ->
+    Tree = concat_texts(Tree0),
+    Changeable = normalize_changeable(filter_changeable(Tree)),
+    ChangeableVars = changeable_vars(Changeable),
+    #{
+        id => changeable_id(State),
+        module => State#state.module,
+        function => State#state.function,
+        static => filter_static(Tree),
+        changeable => Changeable,
+        changeable_vars => group_vars(ChangeableVars),
+        changeable_indexes => changeable_indexes(Changeable),
+        norm_assigns => NormAssigns,
+        norm_assigns_vars => norm_assigns_vars(NormAssigns, ChangeableVars)
+    }.
+
+group_vars(Vars) ->
+    maps:groups_from_list(
+        fun({Var, _}) -> Var end,
+        fun({_, Path}) -> Path end,
+        Vars).
+
+compile_tag(#{name := Name} = Tag, Loc, T, State) ->
+    Attrs = maps:get(attributes, Tag),
+    [{text, <<$<, Name/binary>>} | compile_tag_attrs(Attrs, Tag, Loc, T, State)].
+
+compile_tag_attrs([{Name, {text, _, Value}} | Attrs], Tag, Loc, T, State) ->
+    [{text, <<$\s, Name/binary, $=, $", Value/binary, $">>}
+     | compile_tag_attrs(Attrs, Tag, Loc, T, State)];
+compile_tag_attrs([{Name, {expr, ExprLoc, Expr}} | Attrs], Tag, Loc, T, State) ->
+    Elem = expand_expr(Expr, ExprLoc, true, State),
+    ChangeableCount = case Elem of {text, _} -> 0; {expr, _} -> 1 end,
+    [{text, <<$\s, Name/binary, $=, $">>}, Elem, {text, <<$">>}
+     | compile_tag_attrs(Attrs, Tag, Loc, T, incr_index(ChangeableCount, State))];
+compile_tag_attrs([], #{is_void := true}, _Loc, T, State) ->
+    [{text, <<$\s, $/, $>>>} | compile_elems(T, State)];
+compile_tag_attrs([], #{is_void := false} = Tag, _Loc, T, State) ->
+    InnerContent = maps:get(inner_content, Tag),
+    [{text, <<$>>>} | compile_tag_inner_content(InnerContent, Tag, T, State)].
+
+compile_tag_inner_content(InnerContent, #{name := Name} = Tag, T, State) ->
+    Elems = compile_elems(maybe_push_js_scripts(Name, InnerContent), State),
+    Elems ++ compile_tag_closing(Tag, T, incr_index(changeable_count(Elems, 0), State)).
+
+% We definitely should provide a bundler to inject these scripts.
```

Comment:
This should probably make it to an issue? Otherwise we remove it, if not required "for the now".

Reply:
There is no specific issue for it. The most similar one is https://github.com/arizona-framework/arizona/issues/15. 
I'll remove this comment. We are good to go with the actual implementation.

Reply:
[a11ad98](https://github.com/arizona-framework/arizona/pull/164/commits/a11ad98ba788e6492b79bef125bed0ada96b4200)